### ISSUE TITLE
Undocumented configs for winrm

### ISSUE THREAD

**Issue by javadevmtl** [mitchellh/vagrant#7626]
### Vagrant version

1.8.4
### Host operating system

Ubuntu 14.04
### Guest operating system

Windows Server 2008 with nsidc/vSphere plugin

The VM is launched and sysprep kicks in which causes the Windows VM to take longer to boot and it also reboots a couple times until sysprep is done.

The winrm communicator fails with No route to host, with a retry every 2 seconds based on what is seen from the INFO logs. But the VM in vSphere is up and running coretcly. A second, manual call after checking if sysprep completed, vagrant up call marks the machine as ready.

Google brought me to the original WinRB plugin. config.winrm.max_tries seems to do the trick. None of the "timeout" configs helped.

```
    config.winrm.max_tries = 100 <--- This worked
    config.winrm.timeout = 300 <--- doesn't seem to do anything
```

 config.winrm.max_tries is not documented in the official docs. Maybe it can be documented with a tip indicating what the default is and to bump it up if syspreping?

Thanks

**Comment by dragon788** [comment]
Nice find! The WinRM timeout only affects how long Vagrant waits for the WinRM service (via plugin) to respond before failing.

**Comment by chrisroberts** [comment]
Fixed via #8102
